atmcorr/processing.py

```diff
@@ -43,6 +44,62 @@ def _earthdata_credentials_from_env():
     return auth
 
 
+def _get_tile_index_iter_interpolation(tiling_shape):
+    # make sure to pick corners and center values
+    indices = [(0, 0),
+               (0, tiling_shape[1] - 1),
+               (tiling_shape[0] - 1, 0),
+               (tiling_shape[0] - 1, tiling_shape[1] - 1),
+               (int(tiling_shape[0] / 2), int(tiling_shape[1] / 2))
+               ]
+    # pick every fifth element column/row wise
+    all_indices = list(itertools.product(range(0, tiling_shape[0], 5),
+                                         range(0, tiling_shape[1], 5))) + indices
+    for i in all_indices:
+        yield i
+
+
+def _get_tile_index_iter(tiling_shape):
+    return itertools.product(range(tiling_shape[0]), range(tiling_shape[1]))
+
+
+def _interpolate_correction_params(correction_params, geometry_dict):
+    # interpolate xa and xc
+    for field in ['xa', 'xc']:
```

Comment:
This nested for loop could be something to paralellize easily, if there is a significant speed gain to expect from that.

Reply:
I wont spend more time to paralellize the code here, as we discussed the main bottleneck of the code is the calls to the API. The interpolation part is relatively fast already